docker/Dockerfile.common

```diff
@@ -54,7 +56,32 @@ RUN apt-get -y install \
 RUN python3 -m pip install --upgrade pip
 RUN python3 -m pip install aiofiles debugpy typing_extensions typing
 
+# avahi advertiser support
+RUN apt-get -y install \
+    libnss-mdns \
+    avahi-discover \
+    libavahi-compat-libdnssd-dev
+
 ENV SCRYPTED_DOCKER_SERVE="true"
 ENV SCRYPTED_CAN_RESTART="true"
 ENV SCRYPTED_VOLUME="/server/volume"
 ENV SCRYPTED_INSTALL_PATH="/server"
+
+# copy configurations and scripts
+COPY fs /
+
+# s6 process supervisor
+RUN case "$(uname -m)" in \
```

Comment:
To do that, the `S6_OVERLAY_VERSION` argument will need to be exported to the environment (so that it is available in any scripts called), which will create a new layer when building the Docker image.

In addition, any error messages echoed (such as notifying the user that their architecture isn't supported) won't make their way to the output console, which may lead to a silent fail.

With that being said, I can move it to a shell script anyways, but I wouldn't recommend it. Do you want to leave it as is or move it outside of the Dockerfile?

Reply:
I'm just now realizing that this dockerfile is interpreted by the linux install script. this pull request will cause issues with that. I'll need to deal with that. my first thought is that I put all the s6 overlay stuff into a separate dockerfile prior to common.

Reply:
Alright, edits on this PR are allowed by maintainers. Please let me know when you have figured out isolating s6-overlay from the local Linux install script or if I need to do anything, thanks!